dbt/macros/nearest_pin_neighbors.sql

```diff
@@ -0,0 +1,95 @@
+-- Macro that takes a `source_model` containing PIN geometries and joins it
+-- against itself in order to generate the `num_neighbors` nearest neighbors
+-- for each PIN for each year in the data, where a "neighbor" is defined as
+-- another PIN that is within `radius_km` of the given PIN.
+--
+-- The `source_model` must contain four required columns, and is modeled
+-- after spatial.parcel, which is the primary source for this macro:
+--
+-- * pin10
+-- * year
+-- * x_3435
+-- * y_3435
+{% macro nearest_pin_neighbors(source_model, num_neighbors, radius_km) %}
```

Comment:
This macro is nearly identical to the pre-existing `aws-athena/ctas/proximity-dist_pin_to_pin.sql` query, with three main changes:

1. Parameterization of `source_model`, `num_neigbors`, and `radius_km`
2. Addition of `pin10` to the `distinct_pins` subquery per our missing data discussion on Teams ([diff here](https://github.com/ccao-data/data-architecture/pull/138/commits/8aaf932db4e83748a485727216a7e7b81a5c002c))
3. Reformatting to match dbt styles using `sqlfmt`

Everything else should be identical!